constraintsolver/src/CGSolver.cpp

```diff
@@ -9,29 +9,33 @@
 
 #include <iostream>
 
-namespace alica {
-namespace reasoner {
-using alica::VariableSet;
+namespace alica
+{
+namespace reasoner
+{
+using alica::VariableGrp;
 
 CGSolver::CGSolver(AlicaEngine* ae)
-        : ISolver(ae)
-        , _lastUtil(0.0)
-        , _lastFEvals(0.0)
-        , _lastRuns(0.0) {
+    : ISolver(ae)
+    , _lastUtil(0.0)
+    , _lastFEvals(0.0)
+    , _lastRuns(0.0)
+{
     Term::setAnd(AndType::AND);
     Term::setOr(OrType::MAX);
 }
 
 CGSolver::~CGSolver() {}
 
-bool CGSolver::existsSolutionImpl(const VariableSet& vars, const std::vector<shared_ptr<ProblemDescriptor>>& calls) {
+bool CGSolver::existsSolutionImpl(const VariableGrp& vars, const std::vector<shared_ptr<ProblemDescriptor>>& calls)
+{
     std::shared_ptr<Term> constraint = ConstraintBuilder::TRUE;
     const int dim = vars.size();
 
     auto cVars = make_shared<vector<shared_ptr<autodiff::Variable>>>(dim);
-    std::vector<double> ranges(dim * 2, std::numeric_limits<double>::max());
+    std::vector<double> ranges(dim * 2, std::numeric_limits<double>::max() / 2);
```

Comment:
why / 2 ?

Reply:
it's a stop gap until we can get around to properly redesign the solver.
But basically, it needs to be able to compute (max-min), which would otherwise be infinity.